### PR TITLE
fix: sequelize mode support HAVING, and select fields raw sql support

### PR DESCRIPTION
**Pull request by JimmyDaddy**

### CODE REVIEW

File: src/spell.js [hunk 1]
```diff
@@ -881,6 +883,7 @@ class Spell {
     const { columns, groups } = this;
 
     for (const name of names) {
+      console.log(name, 'name');
```
Comment (cyjake) [inline src/spell.js:886]: console

File: test/unit/adapters/sequelize.test.js [hunk 2]
```diff
@@ -454,6 +454,45 @@ describe('=> Sequelize adapter', () => {
...
+
+    result = await Post.findAll({
+      attributes: 'count(*) AS count',
+      group: [ 'title' ],
+      order: [[ 'title', 'desc' ]],
+    });
+    assert.deepEqual(result, [
+      { title: 'Tyrael', count: 1 },
+      { title: 'Leah', count: 2 },
+    ]);
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:466]: break into a new test case such as `it('Model.findAll({ group: [] })'`?

File: test/unit/adapters/sequelize.test.js [hunk 3]
```diff
@@ -454,6 +454,45 @@ describe('=> Sequelize adapter', () => {
...
+    result = await Post.findAll({
+      attributes: 'count(*) AS count',
+      group: [ 'title' ],
+      order: [[ 'title', 'desc' ]],
+      having: raw('count(*) = 2')
+    });
+
+    assert.deepEqual(result, [
+      { title: 'Leah', count: 2 },
+    ]);
```
Comment (cyjake) [inline test/unit/adapters/sequelize.test.js:495]: same as above, I think breaking into separated test cases helps regression tests and debugging

Comment (cyjake) [reply test/unit/adapters/sequelize.test.js:495]: `having([ 'count = ?', 2 ])` should be supported as well, which can be achieved with `spell.$having(...having)`

File: src/spell.js [hunk 4]
```diff
@@ -69,7 +70,7 @@ function parseObjectValue(value) {
   if (value && value.__raw) return { type: 'raw', value: value.value };
   // value maybe an object conditions
   if (isPlainObject(value)) {
-    const [ args ] = parseObjectConditions(value);
+    const args = parseObjectConditions(value);
     return args;
```
Comment (cyjake) [inline src/spell.js:74]: `return parseObjectConditions(value);`

Comment (cyjake) [reply src/spell.js:74]: this change breaks return type of parseObjectValue however.

File: src/spell.js [hunk 5]
```diff
@@ -273,9 +274,11 @@ function parseObjectConditions(conditions) {
       });
     }
     else if (isLogicalCondition(value)) {
-      const condition = parseObjectValue(value);
-      normalizeLogicalCondition(condition, name);
-      result.push(condition);
+      const conds = parseObjectValue(value);
```
Comment (cyjake) [inline src/spell.js:277]: maybe we can use `parseOperatorCondition(name, value)` here

Comment (cyjake) [reply src/spell.js:277]: tried a bit, with no luck. I'd need more time to investigate on this.

Comment (JimmyDaddy) [reply src/spell.js:277]: `parseOperatorCondition` seems can't parse logical operator and return object not object array

**Pull request merged**